capstone/backend/src/main/java/com/google/spanner/CreateSpannerTables.java

```diff
@@ -0,0 +1,157 @@
+
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.cloud.spanner.Database;
+import com.google.cloud.spanner.DatabaseAdminClient;
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.InstanceAdminClient;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerException;
+import com.google.cloud.spanner.SpannerExceptionFactory;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.spanner.admin.database.v1.CreateDatabaseMetadata;
+import java.util.Arrays;
+import java.util.concurrent.ExecutionException;
+import org.threeten.bp.LocalDate;
+import org.threeten.bp.temporal.ChronoField;
+
+
+public class UsersSpanner {
+
+  /*
+   * Creates database using an Admin Client, accessing the instance signalled in
+   * the DatabaseId and creating 5 hard coded tables: Users, UserLists, Stores, Items,
+   * and Inventory.
+   * 
+   * @param dbAdminClient client used to access the instance and database
+   * @param id contains the information regarding the instance (name, etc.)
+   *
+   * @throws SpannerException if failed during execution
+   */
+  static void createDatabase(DatabaseAdminClient dbAdminClient, DatabaseId id) {
+    OperationFuture<Database, CreateDatabaseMetadata> op =
+        dbAdminClient.createDatabase(
+            id.getInstanceId().getInstance(),
+            id.getDatabase(),
+            Arrays.asList(
+                "CREATE TABLE Users ("
+                    + "  UserId    INT64,"
+                    + "  Username  STRING(MAX),"
+                    + "  Email     STRING(MAX)"
+                    + ") PRIMARY KEY (UserId)",
+                "CREATE TABLE UserLists ("
+                    + "  UserId      INT64,"
+                    + "  ListId      INT64,"
+                    + "  ItemTypes   ARRAY<STRING(MAX)>"
+                    + ") PRIMARY KEY (UserId, ListId),"
+                    + "  INTERLEAVE IN PARENT Users ON DELETE CASCADE",
+                "CREATE TABLE Stores ("
+                    + "  StoreId      INT64,"
+                    + "  StoreName    STRING(MAX),"
+                    + "  Address      STRING(MAX)"
+                    + ") PRIMARY KEY (StoreId)",
+                "CREATE TABLE Items ("
+                    + "  ItemId            INT64,"
+                    + "  ItemNameAndBrand  STRING(MAX),"
+                    + "  ItemType          STRING(MAX)"
+                    + ") PRIMARY KEY (ItemId)",
+                "CREATE TABLE Inventory ("
+                    + "  StoreId           INT64,"
+                    + "  ItemId            INT64,"
+                    + "  ItemAvailability  STRING(MAX),"
+                    + "  LastUpdated       TIMESTAMP OPTIONS (allow_commit_timestamp=true),"
+                    + "  Price             FLOAT64,"
+                    + "  CONSTRAINT FK_ItemId FOREIGN KEY (ItemId) REFERENCES Items (ItemId)"
```

Comment:
This is up to @officialcwang

Reply:
I put that in to enable foreign keys (based on https://cloud.google.com/spanner/docs/foreign-keys/how-to), so that ItemId in the Inventory table would be a foreign key referencing the Items table. I don't think StoreId is a foreign key for the Stores table.